scripts/us_census/pep/population_estimate_by_race/preprocess.py

```diff
@@ -0,0 +1,936 @@
+    df['AmericanIndianAndAlaskaNativeAloneOrInCombinationWithOneOrMore'+\
+        'OtherRaces']= df['IAC_MALE'].astype(int)+df['IAC_FEMALE'].astype(int)
+    df['AsianAloneOrInCombinationWithOneOrMoreOtherRaces'] = df[
+        'AAC_MALE'].astype(int) + df['AAC_FEMALE'].astype(int)
+    df['NativeHawaiianAndOtherPacificIslanderAloneOrInCombinationWithOneOr'+\
+        'MoreOtherRaces']= df['NAC_MALE']\
+            .astype(int)+df['NAC_FEMALE'].astype(int)
+    df.drop(columns=[
+        'AGEGRP', 'TOT_POP', 'TOT_MALE', 'TOT_FEMALE', 'WA_MALE', 'WA_FEMALE',
+        'BA_MALE', 'BA_FEMALE', 'IA_MALE', 'IA_FEMALE', 'AA_MALE', 'AA_FEMALE',
+        'NA_MALE', 'NA_FEMALE', 'TOM_MALE', 'TOM_FEMALE', 'WAC_MALE',
+        'WAC_FEMALE', 'BAC_MALE', 'BAC_FEMALE', 'IAC_MALE', 'IAC_FEMALE',
+        'AAC_MALE', 'AAC_FEMALE', 'NAC_MALE', 'NAC_FEMALE'
+    ],
+            inplace=True)
+    return df
+
+
+def _clean_csv2_file(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    This method cleans the dataframe loaded from a csv file format.
+    Also, Performs transformations on the data.
+
+    Arguments:
+        df (DataFrame) : DataFrame of csv dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for csv dataset.
+    """
+    df.drop(df.index[65:], inplace=True)
+    df.drop(df.index[1:14], inplace=True)
+    modify = [0, 30, 31, 32, 33, 34, 35, 40, 41, 42, 49]
+    for j in modify:
+        df.iloc[j]["Area"] = df.iloc[j]["Area"] + " " + df.iloc[j][1]
+        for i in range(2, 10):
+            df.iloc[j][i - 1] = df.iloc[j][i]
+    df.iloc[9][
+        "Area"] = df.iloc[9]["Area"] + " " + df.iloc[9][1] + " " + df.iloc[9][2]
+    for i in range(3, 11):
+        df.iloc[9][i - 2] = df.iloc[9][i]
+    df.drop(columns=["3", "4", "8", "9", "10"], inplace=True)
+    # Replacing the reuired columns.
+    df.columns = df.columns.str.replace('Area', 'Geographic Area')
+    df.columns = df.columns.str.replace('1', 'Total')
+    df.columns = df.columns.str.replace('2', 'Total White')
+    df.columns = df.columns.str.replace('5', 'Total Black')
+    df.columns = df.columns.str.replace('6', \
+        'Total American Indian & Alaska Native')
+    df.columns = df.columns.str.replace('7', 'Total Asian & Pacific Islander')
+
+    df["Geographic Area"] = [x.title() for x in df["Geographic Area"]]
+    #print(df)
+    return df
+
+
+def _clean_txt2_file(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    This method cleans the dataframe loaded from a txt file format.
+    Also, Performs transformations on the data.
+
+    Arguments:
+        df (DataFrame) : DataFrame of txt dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for txt dataset.
+    """
+    df['1'] = df['1'].astype(str)
+    mask = df['1'].str.len() >= 6
+    df = df.loc[mask]
+    mask = df['2'] >= 50000
+    df = df.loc[mask]
+    mask = df['1'].str[0] == '7'
+    df = df.loc[mask]
+    df = df[df['1'].str.contains("999")]
```

Comment:
Please add functional comment for this block

Reply:
added comment